src/Microsoft.ML.ImageAnalytics/ImageGrayscaleTransform.cs

```diff
@@ -0,0 +1,171 @@
+            {
+                Contracts.AssertValue(sb);
+                return TryUnparseCore(sb);
+            }
+        }
+
+        public class Arguments : TransformInputBase
+        {
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "New column definition(s) (optional form: name:src)", ShortName = "col", SortOrder = 1)]
+            public Column[] Column;
+        }
+
+        internal const string Summary = "Convert image into grayscale.";
+
+        internal const string UserName = "Image Greyscale Transform";
+        public const string LoaderSignature = "ImageGrayscaleTransform";
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "IMGGREY ",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature);
+        }
+
+        private const string RegistrationName = "ImageGrayscale";
+
+        /// Public constructor corresponding to SignatureDataTransform.
+        public ImageGrayscaleTransform(IHostEnvironment env, Arguments args, IDataView input)
+            : base(env, RegistrationName, env.CheckRef(args, nameof(args)).Column, input, t => t is ImageType ? null : "Expected Image type")
+        {
+            Host.AssertNonEmpty(Infos);
+            Host.Assert(Infos.Length == Utils.Size(args.Column));
+            Metadata.Seal();
+        }
+
+        private ImageGrayscaleTransform(IHost host, ModelLoadContext ctx, IDataView input)
+            : base(host, ctx, input, t => t is ImageType ? null : "Expected Image type")
+        {
+            Host.AssertValue(ctx);
+            // *** Binary format ***
+            // <base>
+            Host.AssertNonEmpty(Infos);
+            Metadata.Seal();
+        }
+
+        public static ImageGrayscaleTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var h = env.Register(RegistrationName);
+            h.CheckValue(ctx, nameof(ctx));
+            h.CheckValue(input, nameof(input));
+            ctx.CheckAtModel(GetVersionInfo());
+            return h.Apply("Loading Model", ch => new ImageGrayscaleTransform(h, ctx, input));
+        }
+
+        public override void Save(ModelSaveContext ctx)
+        {
+            Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            // *** Binary format ***
+            // <base>
+            SaveBase(ctx);
+        }
+
+        protected override ColumnType GetColumnTypeCore(int iinfo)
+        {
+            Host.Assert(0 <= iinfo & iinfo < Infos.Length);
+            return Infos[iinfo].TypeSrc;
+        }
+
+        public ColorMatrix GreyscaleColorMatrix = new ColorMatrix(
```

Comment:
>public ColorMatrix GreyscaleColorMatrix [](start = 8, length = 39)

This is a mutable, public instance level field that affects the functionality of the transform.

Probably want this to be `private` (or at least `internal` if somehow it is referenced somewhere else), `static`, and almost certainly `readonly`.